src/request.ts

```diff
@@ -1186,6 +1187,48 @@ class DatastoreRequest {
     this.save(entities, callback);
   }
 
+  merge(entities: Entities): Promise<CommitResponse>;
+  merge(entities: Entities, callback: CallOptions): void;
+  /**
+   * Maps to {@link Datastore#save}, forcing the method to be `merge`.
+   *
+   * @param {object|object[]} entities Datastore key object(s).
+   * @param {Key} entities.key Datastore key object.
+   * @param {string[]} [entities.excludeFromIndexes] Exclude properties from
+   *     indexing using a simple JSON path notation. See the examples in
+   *     {@link Datastore#save} to see how to target properties at different
+   *     levels of nesting within your entity.
+   * @param {object} entities.data Data to merge to the same for provided key.
+   * @param {function} callback The callback function.
+   * @param {?error} callback.err An error returned while making this request
+   * @param {object} callback.apiResponse The full API response.
+   */
+  merge(
+    entities: Entities,
+    callback?: CallOptions
+  ): void | Promise<CommitResponse> {
+    arrify(entities)
```

Comment:
I believe we should have this in a Transaction, so that the entity is locked while we are getting/updating.

Reply:
Here's an example: https://github.com/googleapis/nodejs-datastore/blob/6b9ce7712af06f4a5e9c2bc2cc9f8c3ef31f2aa5/samples/concepts.js#L1080-L1100